omnichain/swap/contracts/SwapToAnyToken.sol

```diff
@@ -59,16 +70,25 @@ contract SwapToZeta is zContract, OnlySystem {
         uint256 outputAmount = SwapHelperLib.swapExactTokensForTokens(
             systemContract,
             zrc20,
-            isTargetZeta ? amount : amount - inputForGas,
-            target,
+            params.withdraw ? amount - inputForGas : amount,
+            params.target,
             0
         );
 
-        if (isTargetZeta) {
-            IWETH9(wzeta).transfer(address(uint160(bytes20(to))), outputAmount);
+        if (params.withdraw) {
+            IZRC20(gasZRC20).approve(params.target, gasFee);
+            IZRC20(params.target).withdraw(params.to, outputAmount);
         } else {
-            IZRC20(gasZRC20).approve(target, gasFee);
-            IZRC20(target).withdraw(to, outputAmount);
+            IWETH9(params.target).transfer(address(uint160(bytes20(params.to))), outputAmount);
         }
     }
+
+    function bytesToBool(bytes calldata data, uint256 offset)
+        internal
+        pure
+        returns (bool)
+    {
+        require(offset < data.length, "Offset is out of bounds");
```

Comment:
move to custom error

Reply:
This function will be removed and imported from the toolkit:

https://github.com/zeta-chain/toolkit/blob/d952cd7b67526c218e07030263d0d5413ab0e64c/contracts/BytesHelperLib.sol#L54-L63